src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/WithType.cs

```diff
@@ -74,32 +75,21 @@ public new AggregateType GetType()
             return swt1.Sym == swt2.Sym && swt1.Ats == swt2.Ats;
         }
 
-        public static bool operator !=(SymWithType swt1, SymWithType swt2)
-        {
-            if (ReferenceEquals(swt1, swt2))
-            {
-                return false;
-            }
-            else if (ReferenceEquals(swt1, null))
-            {
-                return swt2._sym != null;
-            }
-            else if (ReferenceEquals(swt2, null))
-            {
-                return swt1._sym != null;
-            }
-            return swt1.Sym != swt2.Sym || swt1.Ats != swt2.Ats;
-        }
+        public static bool operator !=(SymWithType swt1, SymWithType swt2) => !(swt1 == swt2);
 
+        [ExcludeFromCodeCoverage] // == overload should always be the method called.
         public override bool Equals(object obj)
         {
+            Debug.Fail("Sub-optimal equality called. Check if this is correct.");
             SymWithType other = obj as SymWithType;
             if (other == null) return false;
             return Sym == other.Sym && Ats == other.Ats;
```

Comment:
This is not actually the same as `==`. It's unclear if this is a bug or not, though it doesn't matter in practice given that it's never called.